dolphin/tests/unit/alert_manager/test_alert_processor.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2020 The SODA Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+from unittest import mock
+from oslo_utils import importutils
+
+from dolphin import context
+from dolphin import exception
+from dolphin.tests.unit.alert_manager import fakes
+
+
+class AlertProcessorTestCase(unittest.TestCase):
+    ALERT_PROCESSOR_CLASS = 'dolphin.alert_manager.alert_processor' \
+                            '.AlertProcessor'
+
+    def _get_alert_processor(self):
+        alert_processor_class = importutils.import_class(
+            self.ALERT_PROCESSOR_CLASS)
+        alert_processor = alert_processor_class()
+        return alert_processor
+
+    @mock.patch('dolphin.db.storage_get')
+    @mock.patch('dolphin.drivers.api.API.parse_alert')
+    @mock.patch('dolphin.alert_manager.alert_processor.AlertProcessor'
+                '._export_alert_model')
+    def test_process_alert_info_success(self, mock_export_model,
+                                        mock_parse_alert, mock_storage):
+        fake_storage_info = fakes.fake_storage_info()
+
+        input_alert = {'storage_id': 'abcd-1234-56789',
```

Comment:
input_alert contains info of SYmmetrix. 
Then below you assign it to expected_driver_input and change the relevant vendor specific things..
What is the need of input_alert? 
Please explain

Reply:
process_alert_info function is supposed to fill some fields using the input_alert and call driver and driver is supposed to give a model out based on input info. U can see that driver_input and driver_output fields are checked  for  correctness
input_alert if passed to 
alert_processor_inst.process_alert_info(input_alert)